enhancements/installer/bootstrapping-clusters-with-capi-providers.md

```diff
@@ -0,0 +1,683 @@
+Changes would need to be made to both the machine-api and cluster-api manifests.
+
+#### External/Upstream dependencies
+
+As explained above, the Installer will embed binaries for each supported CAPI provider, as well
+as `etcd` and `kube-apiserver` binaries. Initially, CAPI providers will be vendored and built in
+the Installer repo. `etcd` and `kube-apiserver` will be copied from the release image.
+
+##### In-tree Dependency Vendoring
+
+For the initial implementation, CAPI providers will be vendored and built within the Installer repo:
+
+```bash
+$ tree cluster-api/providers -L 2
+cluster-api/providers
+├── aws
+│   ├── go.mod
+│   ├── go.sum
+│   ├── tools.go
+│   └── vendor
+├── azure
+│   ├── go.mod
+│   ├── go.sum
+│   ├── tools.go
+│   └── vendor
+...
+```
+
+This follows a similar pattern to the current Terraform provider implementation and build. This
+pattern will be used initially due to its simplicity and existing support. This pattern has
+drawbacks because changes to providers need to be merged upstream first and then vendored
+to the Installer. This aspect will be particularly problematic once CAPI providers are GA
+and we need to keep the Installer in-sync with other providers. Builds are also inefficient
+in that providers are always rebuilt, even when unchanged.
+
+##### Copying Dependencies from Container Images
+
+The previously described vendoring pattern presents disadvantages:
+* increase in the repo size;
+* security issues/warnings in code we do not own;
+* providers are rebuilt with every Installer change, whether or not the change is pertinent to the provider
+
+So as not to incur the same disadvantages, we opted to use a different approach for etcd and kube-apiserver needed for
+the provisioning via CAPI: they will be built into intermediary container images. During the Installer container build,
+those images are imported and the relevant binaries extracted. Because of that, etcd and kube-apiserver will have build
+cadences decoupled from the Installer's, being rebuilt only when needed.
+Because they run in the host machine, like the Installer, they will have to be built for all systems/arches supported,
+including cross-compiled darwin amd64/arm64.
+
+This architecture has 2 opposing disadvantages:
+* In CI: because there is no mechanism to detect and rebuild dependant images, Installer and etcd/KAS might temporarily
+drift apart and cause job failures when the expected versions do not match. This issue is probably alleviated by the
+frequent changes to the Installer repo and can be worked around by forcing an Installer rebuild.
+* In the release: in this case, there is a mechanism to trigger rebuilds automatically. Which means that the Installer
+image will be rebuilt with any changes to etcd/KAS, even if those changes don't necessarily require an Installer rebuild.
+
+For FIPS supported architectures the dependencies will be copied from the release image, while others will
```

Comment:
During build time or at runtime? And I guess we mean KAS/etcd as the dependencies here, right?

FIPS supported architectures basically means Linux amd64, s390x, and ppc64le (arm64 HW doesn't support fips). We are building Linux/amd64 as part of the intermediary images, so the only binaries we need from the release image are the native s390x and ppc64le ones.